content/tutorials/react-native-presence-publish-subscribe.textile

```diff
@@ -0,0 +1,539 @@
+
+leaveChat = () =>{
+  AsyncStorage.removeItem("user");
+  channel.presence.leave();
+  this.setState({"user": ''});
+  this.props.handler();
+}
+
+  render() {
+    chatMessages = this.load_messages();
+    return (
+      <View style={styles.container}>
+      <View style={{
+           alignItems: 'center', flexDirection: 'row', top: 0,marginBottom:5
+        }}>
+        <Button
+            style={{ width: '40%' }}
+            onPress={this.leaveChat}
+            title="Leave Chat"
+            color="#841584" />
+        </View>
+        <ScrollView >
+
+          {chatMessages}
+
+        </ScrollView>
+        <View style={{
+          flexWrap: 'wrap', alignItems: 'flex-start', flexDirection: 'row',
+          position: 'absolute', bottom: 0}}>
+          <TextInput
+            value={this.state.txt}
+            style={{ width: '80%' }}
+            placeholder="Enter Your message!"
+            disabled={this.state.user==''}
+            onChangeText={(txt) => this.setState({ txt })}
+          />
+
+          <Button
+            style={{ width: '20%' }}
+            onPress={this.submitChat}
+            title="Send"
+            color="#841584" />
+        </View>
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: '#F5FCFF',
+    paddingTop: 10,
+  },
+});
+```
+If you look at the code above, these are the differences between the former @chat.js@ file and this new one.
+
+Above, you have imported React Native UI component (View, Text, Button, TextInput, StyleSheet, ScrollView) to help us with our UI styling. You imported @AsyncStorage@, which you will use to store and retrieve items on the device. Also, you required the @ably@ real-time library, and declared two new variables namely @ably@ and @channel@.
+
+In the constructor, you declared a constructor passing in @props@, called the super constructor of the @props@. You declared three items in your state assignment namely @msg@ , @txt@ and @user@.
+
+Next, you defined the @componentDidMount@ method which @React@ will fire once the component is mounted. In this method, you used the @AsyncStorage@ component to check if a value exists and if it does, set the state of @user@ to that value. Also, you called the @subscribe@  function which we will treat soon.
+
+The @subscribe@ method is where all real-time subscriptions such as @presence@ and @subscribe@ are made. Let’s look at what goes on in this method.
+
+# First, you set the @ably@ variable to an instance of the real-time library passing in your API Key and the current user. The reason for putting in the current user is because you want to access presence events.
+# You made a subscription to a channel called @cog-tie@.
+# You made a subscription to the presence event of @enter@ which notifies that a new member has joined the chat (by calling @channel.presence.subscribe@). You then created a message object from the response and append to the @msg@ state.
```

Comment:
`You made a subscription` => `You subscribed to @enter@ presence events by registering a callback that is invoked whenever a new member has entered the channel`

Reply:
Fixed, thanks